### PR TITLE
test(cli): add tests for fsmode generator package

### PR DESCRIPTION
**Pull request by VajiraPrabuddhaka**
## Purpose

Add tests for the `internal/occ/fsmode/generator` package to improve confidence in componentrelease and releasebinding generation logic.

## Approach

**release_binding_test.go:**
- Add tests for `GenerateBinding` validation errors (missing project, component, env, pipeline, namespace)
- Add tests for `selectComponentRelease` with explicit release name (found and not-found cases)
- Add tests for non-root environment promotion path (promote from previous env, no binding in previous env, empty releaseName in previous binding)
- Add tests for `GenerateBulkBindings` (all components, by project name, partial errors, validation errors)

**component_release_test.go:**
- Add test for project name mismatch error
- Add test for unsupported trait kind error
- Add test for component with parameters (verifies componentProfile.parameters)
- Add test for duplicate trait deduplication (same trait, different instance names)

## Related Issues

N/A

## Checklist
- [x] Tests added or updated (unit, integration, etc.)
- [ ] Samples updated (if applicable)

### CODE REVIEW

File: internal/occ/fsmode/generator/release_binding_test.go [hunk 1]
```diff
@@ -356,6 +357,360 @@ func TestSelectComponentRelease_CompareSpecsError(t *testing.T) {
...
+		result, err := gen.GenerateBulkBindings(BulkBindingOptions{
+			ProjectName:  "proj-a",
+			TargetEnv:    "dev",
+			PipelineInfo: pipelineInfo,
+			Namespace:    namespace,
+		})
+		require.NoError(t, err)
+		assert.Len(t, result.Bindings, 2)
+		assert.Empty(t, result.Errors)
+	})
```
[the remark this answers is not in the document]
Comment (VajiraPrabuddhaka) [reply internal/occ/fsmode/generator/release_binding_test.go:677]: No, this test case only covers a particular project. I think to address your concern, we have to add another test case to cover multiple projects.

[the remark this answers is not in the document]
Comment (VajiraPrabuddhaka) [reply internal/occ/fsmode/generator/release_binding_test.go:677]: I have done a separate test case for this.

**Review by LakshanSS — APPROVED** [review]
